packages/web-twig/src/Twig/PropsExtension.php

```diff
@@ -128,6 +176,24 @@ public function useStyleProps(array $props): array
         $styleProps['className'] = $props['UNSAFE_className'] ?? null;
         $styleProps['style'] = $props['UNSAFE_style'] ?? null;
 
+        foreach ($props as $propName => $propValue) {
+            if (array_key_exists($propName, self::SPACING_STYLE_PROPS)) {
+                if (is_array($propValue)) {
+                    foreach ($propValue as $breakpoint => $value) {
+                        $utilityName = self::SPACING_STYLE_PROPS[$propName];
+                        $utilityValue = $value == 'auto' ? 'auto' : preg_replace('/[^0-9]/', '', $value);
+                        $infix = $breakpoint === 'mobile' ? '' : '-' . $breakpoint;
+                        $styleProps['className'] .= ' ' . $classPrefix . $utilityName . $infix . '-' . $utilityValue;
```

Comment:
You can use a function like `sprintf` (https://www.php.net/manual/en/function.sprintf.php) to format the output. This format is used on two places, so maybe add a new method for this?

Reply:
Maybe all these lines around `if (is_array($propValue))` can be refactored with more functional approach so it's more readable and DRY. For example:

```php
$spacingClasses = [];

if (is_array($propValue)) {
  foreach ($propValue as $breakpoint => $value) {
    array_push($spacingClasses, getSpacingClassName($propName, $value, $breakpoint));
  }
} else {
  array_push($spacingClasses, getSpacingClassName($propName, $propValue));
}

if (count($spacingClasses)) {
  $styleProps['className'] .= self:CLASS_SEPARATOR . implode(self:CLASS_SEPARATOR, $spacingClasses);
}
```

@literat is surely a better person to polish (or reject) the idea 🙂.

Reply:
see updated code